### DISCUSSION TITLE
Storage Error: RPC failure, probably timeout.

### DISCUSSION THREAD

**Discussion by salahaz** [vesoft-inc/nebula#5695] [Q&A]
When using nebula-python to execute queries it throws:
` Storage Error: RPC failure, probably timeout.`

This happens even though, the query is optimized using LIMIT. Here's the query:
`match (n:malware) return n.malware.name limit 3;`

Mind you, running the exact same query in Nebula console works.

NebulaGraph is run using the nebula-docker-compose, its version is 3.6.0

All cluster containers are running and healthy; the logs don't log any specific errors.

Any ideas?

**Comment by QingZ11** [comment]
The statement `match (n:malware) return n.malware.name limit 3;` will scan all the data, which can easily lead to resource exhaustion. It is recommended to create an index on the `malware` tag for efficient querying. Utilizing index-based queries will save computational resources.
